GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,246 @@
+
+        protected override string VerbName
+        {
+            get { return HealthVerbName; }
+        }
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            // The path to the root of git's tree
+            string sourceRoot = enlistment.WorkingDirectoryRoot;
+
+            if (this.Directory == null)
+            {
+                this.Directory = string.Empty;
+            }
+            else if (this.Directory.Equals("."))
+            {
+                this.Directory = Environment.CurrentDirectory.Substring(sourceRoot.Length);
+            }
+
+            // Get all of the data needed to calculate statistics then pass them into GVFSEnlistmentStatistics
+            List<string> modifiedPathsFileList = new List<string>();
+            List<string> modifiedPathsFolderList = new List<string>();
+            List<string> placeholderFilePathList = new List<string>();
+            List<string> placeholderFolderPathList = new List<string>();
+
+            this.GetPlaceholdersFromDatabase(enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders);
+            foreach (IPlaceholderData placeholderData in filePlaceholders)
+            {
+                placeholderFilePathList.Add(placeholderData.Path);
+            }
+
+            foreach (IPlaceholderData placeholderData in folderPlaceholders)
+            {
+                placeholderFolderPathList.Add(placeholderData.Path);
+            }
+
+            foreach (string path in this.GetModifiedPathsFromPipe(enlistment))
+            {
+                if (path.Last() == GVFSConstants.GitPathSeparator)
+                {
+                    path.TrimEnd('/');
+                    modifiedPathsFolderList.Add(path);
+                }
+                else
+                {
+                    modifiedPathsFileList.Add(path);
+                }
+            }
+
+            this.GetPathsFromGitIndex(enlistment, out List<string> gitFilePaths, out List<string> gitFolderPaths, out List<string> skipWorkTreeFiles);
+
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData pathData = new GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData(
+                gitFolderPaths,
+                gitFilePaths,
+                placeholderFolderPathList,
+                placeholderFilePathList,
+                modifiedPathsFolderList,
+                modifiedPathsFileList,
+                skipWorkTreeFiles);
+            GVFSEnlistmentHealthCalculator enlistmentHealthCalculator = new GVFSEnlistmentHealthCalculator(pathData);
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData = enlistmentHealthCalculator.CalculateStatistics(this.Directory, this.DirectoryDisplayCount);
+
+            this.PrintOutput(enlistment, enlistmentHealthData);
+        }
+
+        private void PrintOutput(
+            GVFSEnlistment enlistment,

```

Comment:
Do we still need this parameter?

Reply:
Nope, I removed it. It became unused after the changes yesterday